### PR TITLE
Integration with Media Session API

### PR DESCRIPTION
**Pull request by idashevskii**
Tested with XFCE pulseaudio plugin.
image

### CODE REVIEW

File: README.md
Comment (vrizo) [inline README.md:29]: Could you please explain me what is a multimedia keyboard? Do you mean multimedia keys on a keyboard?

File: source/controls.js
Comment (vrizo) [inline source/controls.js]: Oh, sorry, looks like I set up linter incorrectly. Could you please delete `;` on the end of the line everywhere? to be consistent with the project

File: source/controls.js
Comment (vrizo) [inline source/controls.js]: Please replace `const` with `let`. It’s OK to declare `const` on the top level scope, but it might be confusing inside of methods, because it’s not a true constant. I’ll try to find the article I refer

**Review by vrizo — CHANGES_REQUESTED** [review]
Hi, @idashevskii !

Looks great, thanks a lot! I didn’t tested the code locally yet but I think it works as intended. Please check my feedback and I think we will be ready to merge the PR after a few changes.

Thank you!

File: README.md
Comment (idashevskii) [reply README.md:29]: > Do you mean multimedia keys on a keyboard?

Yep, exactly. The idea is to control YaMusic while the browser is not focused or even minimized. Tt is the easiest way to do that I found.
Also it integrates with operation system and allows you control YaMusic even without keyboard.

https://developer.mozilla.org/en-US/docs/Web/API/Media_Session_API

For example, in windows (pic from Google):
image

In XFCE (my screenshot):
image

File: source/controls.js
Comment (idashevskii) [reply source/controls.js]: Fixed

File: source/controls.js
Comment (idashevskii) [reply source/controls.js]: Fixed

**Comment by idashevskii** [comment]
@vrizo thank you for review, the formatting was changed to fit specified requirements

File: README.md
Comment (vrizo) [reply README.md:29]: Looks great, thanks a lot! I think we need to tell about this feature to users on Settings page instead of `Unfortunately, Firefox doesn’t support multimedia keys yet`.

Also we can add a hint:

hint screenshot

I think it doesn’t block merging, and we can add it later in the next release if you wish. Users ask for global hotkeys sometimes, and it might help.

**Comment by vrizo** [comment]
@idashevskii, amazing work! It is extremely useful integration and code looks perfect! I think we’re ready to merge. I’m going to test it tomorrow, add a couple of changes to README and send it to Mozilla Addons. Thank you!

File: README.md
Comment (idashevskii) [reply README.md:29]: > Also we can add a hint:

Do you mean that if user's platform doesn't support Media Session API, he should get notification about that?

Comment (vrizo) [reply README.md:29]: Nope, I mean, we can show a notification that the plugin supports Media Session API now. But the user most likely doesn’t know what is Media Session API.

I’ve tested the plugin on Mac today and I think now that the hint is useless because global hotkeys work out-of-the-box. I supposed that it requires to install some software but it’s not necessary.

**Pull request merged**
